### PR TITLE
Versions in java files

### PR DESCRIPTION
**Pull request by JimSuplizio**
This is the same as the previous PR that was cancelled on Friday due to the DevOps pipeline being down.

### CODE REVIEW

File: sdk/storage/azure-storage-blob-cryptography/pom.xml [hunk 1]
```diff
@@ -70,13 +70,13 @@
       cross platform compatibility within SDKs -->
       <groupId>com.microsoft.azure</groupId>
       <artifactId>azure-storage</artifactId>
-      <version>8.3.0</version>
+      <version>8.4.0</version>
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob-cryptography/pom.xml]: @gapra-msft @rickle-msft is it okay to change these versions?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -157,3 +156,32 @@ This will go through the entire source tree and update all of the references in
...
...o a library that also requires dependency changes:** Right now things are in a state where dependency changes can be made along with libraries that depend on them because of the project dependencies in the service pom files. Local development isn't going to change that much except when changing the version of a library and its dependency or dependencies means that the service poms are going to have to be built, and installed, in the appropriate order. This is because these new versions of the library dependencies won't yet be released and Maven will need to find these in the local cache. The biggest change to the process is going to be around PRs and publishing. Separate PRs going going to have to be submitted in order, with dependencies being submitted first. This is necessary because the dependencies need to be published in order to allow things that depend on them to continue using the published version. Trying to submit everyting in one PR would cause build breaks since the dependency being referenced is a version not yet published. An example of this would be making changes to `com.azure:azure-storage-common` that also required dependency changes to `com.azure:azure-identity`.
```
Comment (weshaggard) [inline CONTRIBUTING.md]: "going going" -> "are going"
"everting" -> "everything"

**Pull request merged**